### DISCUSSION TITLE
Python應用於深度學習

### DISCUSSION THREAD

**Discussion by enoch-lightning** [1101-GS4719-C/Python-Community-GS4719-C#205] [Q&A]
目前想使用Python操作深度學習(deep learning)，主題是要從照片中區分出男生與女生。使用的資料集為python的tensorflow的celeb_a。我有找到別人分享使用深度學習中的捲積類神經網路(CNN，convolutional neural network)，結合celeb_a資料集(dataset)來分辨是男生還是女生的文章。我一直想從tensorflow import celeb_a到google colab，但怎麼試就是不成功。
文章資料如這邊所附。
[Gender Inference with Deep Learning](https://towardsdatascience.com/gender-identification-with-deep-learning-ac379f85a790)
此篇文章有對應的GitHub儲存庫，連結如下所附。
[文章作者GitHub儲存庫](https://github.com/njermain/GenderID-ConvNet)
[情況解說影片](https://youtu.be/jyvd_zdcme4)

**Comment by luo910227** [comment]
有看了你的影片，我在對於Tensorflow Dataset的extract大概了解之下、看了此網站提到的然後結合你有提到程式錯誤時有關於GIB記憶體容量的問題，小弟猜測有可能是因記憶體跟效能的問題導致他們原本也是使用壓縮檔?
我只是試著查閱資料發想可能原因，若有不合理的思考麻煩跟我說明(⁎˃ᆺ˂)

網站文章結論：我們在實作練習時，常一次載入所有資料，並未發生問題，但是實際執行專案時，就必須考慮效能，Tensorflow Dataset 是一個不錯的模組，不僅可以節省記憶體，也可以提升效能。

https://ithelp.ithome.com.tw/articles/10241789

**Reply by enoch-lightning** [reply]
我原本的想法沒有想到與效率有關，我是想說難道celeb_a的資料集，一定要下載下來，再上傳到google colab嗎？因為像是手寫辨識的mnist只要使用`from tensorflow.keras.datasets import mnist`就可匯入`
tensorflow.keras.datasets`的屬性mnist，也就是mnist資料集(dataset)。
